### ISSUE TITLE
Auto detect and warn against performance pitfalls

### ISSUE THREAD

**Issue by avik-pal** [LuxDL/Lux.jl#699]
## Implementation

Set up a preferences-based system for doing the following:

1. `default`: Do nothing, i.e., the current state
2. `warn`: If we detect type mismatches, we display a warning and tell users to use `Experimental.@debug_mode` to figure out the exact location of the model. (I don't mind having this as the default but this will be breaking if we choose to do so)
3. `autoconvert`: My least favorite version (Flux's default), but few people do seem to like this. We convert the types to the most performant version.
4. `error`: Throw an error if eltype mismatch is detected.

## Documentation

1. Document the preferences for eltype detection
2. Create a Page showcasing the various options, even from GPUArrays verse
3. Options from `DispatchDoctor` once that is merged into LuxCore
